src/libraries/Microsoft.Extensions.DependencyInjection.Specification.Tests/src/KeyedDependencyInjectionSpecificationTests.cs

```diff
@@ -158,6 +159,40 @@ public void ResolveKeyedServicesAnyKeyWithAnyKeyRegistration()
             Assert.Equal(new[] { service1, service2, service3, service4 }, allServices.Skip(1));
         }
 
+        [Fact]
+        public void ResolveKeyedServicesAnyKeyConsistency()
+        {
+            var serviceCollection = new ServiceCollection();
+            var service = new Service("first-service");
+            serviceCollection.AddKeyedSingleton<IService>("first-service", service);
+
+            var provider1 = CreateServiceProvider(serviceCollection);
+            Assert.Null(provider1.GetKeyedService<IService>(KeyedService.AnyKey));
+            Assert.Equal(new[] { service }, provider1.GetKeyedServices<IService>(KeyedService.AnyKey));
+
+            var provider2 = CreateServiceProvider(serviceCollection);
```

Comment:
This section is identical to the one above - is it verifying that caching doesn't break the second case?

Reply:
Exaclty

Reply:
I added some comments in the test.